x/paramfilter/test/param_filter_test.go

```diff
@@ -1,63 +1,119 @@
-	for _, p := range app.BlockedParams() {
-		// try to set the max validators to 2, unlike above this should fail
-		validChange := proposal.NewParamChange(stakingtypes.ModuleName, string(stakingtypes.KeyMaxValidators), "2")
-		invalidChange := proposal.NewParamChange(p[0], p[1], "value")
-		p := testProposal(validChange, invalidChange)
-		err := handler(ctx, p)
+		// Ensure that the validator params still haven't been modified
+		require.Equal(t, defaults, *testApp.GetConsensusParams(ctx).Validator)
+	})
+
+	t.Run("test that another proposal with a blocked param (evidence consensus params) is rejected", func(t *testing.T) {
+		defaults := coretypes.DefaultEvidenceParams()
+		// Ensure that the evidence params haven't been modified yet
+		require.Equal(t, defaults, *testApp.GetConsensusParams(ctx).Evidence)
+
+		updated := tmproto.EvidenceParams{
+			MaxAgeNumBlocks: defaults.MaxAgeNumBlocks + 1,
+			MaxAgeDuration:  1,
+			MaxBytes:        defaults.MaxBytes,
+		}
+		require.NoError(t, baseapp.ValidateEvidenceParams(updated))
+
+		marshalled, err := testApp.AppCodec().MarshalJSON(&defaults)
+		require.NoError(t, err)
+
+		invalidChange := proposal.NewParamChange(baseapp.Paramspace, string(baseapp.ParamStoreKeyEvidenceParams), string(marshalled))
+		p := testProposal(invalidChange)
+		err = handler(ctx, p)
 		require.Error(t, err)
 		require.Contains(t, err.Error(), "parameter can not be modified")
 
-		ps := app.StakingKeeper.GetParams(ctx)
-		require.Equal(t, ps.MaxValidators, uint32(1))
+		// Ensure that the evidence params still haven't been modified
+		require.Equal(t, defaults, *testApp.GetConsensusParams(ctx).Evidence)
+	})
 
-	}
+	t.Run("test that a proposal with an unblocked params is accepted", func(t *testing.T) {
+		// Ensure that MaxValidators has not been modified
+		require.Equal(t, stakingtypes.DefaultMaxValidators, testApp.StakingKeeper.GetParams(ctx).MaxValidators)
+
+		newMaxValidators := uint32(1)
+		validChange := proposal.NewParamChange(stakingtypes.ModuleName, string(stakingtypes.KeyMaxValidators), fmt.Sprint(newMaxValidators))
+
+		p := testProposal(validChange)
+		err := handler(ctx, p)
+		require.NoError(t, err)
+		require.Equal(t, newMaxValidators, testApp.StakingKeeper.GetParams(ctx).MaxValidators)
+	})
+
+	t.Run("test that a proposal with a blocked param and an unblocked param is rejected", func(t *testing.T) {
+		for _, p := range testApp.BlockedParams() {
+			ps := testApp.StakingKeeper.GetParams(ctx)
+			// Ensure that MaxEntries has not been modified
+			require.Equal(t, stakingtypes.DefaultMaxEntries, ps.MaxEntries)
+
+			newMaxEntries := 8
+			validChange := proposal.NewParamChange(stakingtypes.ModuleName, string(stakingtypes.KeyMaxEntries), fmt.Sprint(newMaxEntries))
+			invalidChange := proposal.NewParamChange(p[0], p[1], "value")
+
+			p := testProposal(validChange, invalidChange)
+			err := handler(ctx, p)
+			require.Error(t, err)
+			require.Contains(t, err.Error(), "parameter can not be modified")
```

Comment:
If it could be converted to an error type, it would enhance code readability and make potential future refactoring easier.

Reply:
hmm it's defined here: https://github.com/rootulp/celestia-app/blob/90b761bb7c2918e0ca75fc488801cd9b914431c5/x/paramfilter/types.go#L15

idk if it can be "converted to an error type" b/c 

```
// Register returns an error instance that should be used as the base for
// creating error instances during runtime.
```

but I can replace the error assertions with this diff if you prefer:

```diff
-			require.Contains(t, err.Error(), "parameter can not be modified")
+			require.ErrorIs(t, err, paramfilter.ErrBlockedParameter)
```

Reply:
Attempted to address via https://github.com/celestiaorg/celestia-app/pull/2453/commits/4c452c5f4dc9de1225f91ea2618c6fca79297f44. Please unresolve if that doesn't address your concern.